src/jit/importer.cpp

```diff
@@ -240,7 +240,7 @@ bool Compiler::impILConsumesAddr(const BYTE* codeAddr, CORINFO_METHOD_HANDLE fnc
             var_types lclTyp = JITtype2varType(info.compCompHnd->getFieldType(resolvedToken.hField, &clsHnd));
 
             // Preserve 'small' int types
-            if (lclTyp > TYP_INT)
+            if (genTypeSize(lclTyp) > genTypeSize(TYP_INT))
```

Comment:
Here too it's not about the type size. The condition should probably be `if (!varTypeIsSmall(lclType))`. It happens so that `lclTyp > TYP_INT` has the same effect because `genActualType(TYP_INT)` is `TYP_INT`.

Same for the other 2 cases below.

Reply:
Thank you, will fix.